### PR TITLE
Avoid to import matplotlib to set its backend Agg in code like chainer

### PR DESCRIPTION
**Pull request by vilyaair**
Refs https://github.com/chainer/chainer/pull/3043

### CODE REVIEW

**Comment by okuta** [comment]
This PR is blocked by test script issue.

File: examples/gmm/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+### How to demo
+The demo contains a script that partitions data into groups using Gaussian Mixture Model.
+The demo can be run by the following command.
+
+```
+python gmm.py [--gpu-id GPU_ID] [--num NUM] [--dim DIM]
+              [--max-iter MAX_ITER] [--tol TOL] [--output-image OUTPUT]
+```
+
+If you run this script on Linux, setting the environmental variable `MPLBACKEND` to `Agg` may be required to use `matplotlib`. For example,
```
Comment (kmaehashi) [inline examples/gmm/README.md]: `on Linux` -> `on non-GUI environment`?

File: examples/kmeans/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+The demo contains a script that partitions data into groups using K-means clustering.
+The demo can be run by the following command.
+
+```
+python kmeans.py [--gpu-id GPU_ID] [--n-clusters N_CLUSTERS] [--num NUM]
+                 [--max-iter MAX_ITER] [--use-custom-kernel]
+                 [--output-image OUTPUT_IMAGE]
+```
+
+If you run this script on Linux, setting the environmental variable `MPLBACKEND` to `Agg` may be required to use `matplotlib`. For example,
```
Comment (kmaehashi) [inline examples/kmeans/README.md]: ditto

**Comment by okuta** [comment]
Thank you for your reviewing.

@vilyaair Cloud you fix?

**Comment by vilyaair** [comment]
Definitely, my pleasure.
Initially, I want to keep the `README.md ` more consistent with [that](https://github.com/chainer/chainer/blob/master/examples/vae/README.md) in the chainer. I think that neither `Linux` nor `non-GUI environment` is a precise term. A typical situation is Python embedded distribution on Windows, which does not contain TkInter. Therefore, it also requires to set the environment variable `MPLBACKEND` to `Agg`. I feel that it is more accurate to use [`environment without appropriate renderers`](https://matplotlib.org/users/installing.html#dependencies), but it is not intuitive.

**Comment by kmaehashi** [comment]
@vilyaair Thanks for the insights.
How about saying like `on environment without matplotlib renderers (e.g., non-GUI environment)`?

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by okuta** [comment]
@kmaehashi The test was passed.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**